src/haz3lweb/view/ScratchMode.re

```diff
@@ -98,7 +98,180 @@ let download_slide_state = state => {
+                | _ => Uuidm.nil
+                }
+              | _ => Uuidm.nil
+              }
+            | None => Uuidm.nil
+            }
+          },
+          ancestors,
+        );
+      let view_fun = (name, ids, level) => [
+        div(
+          ~attr=
+            Attr.many([
+              clss(["breadcrumb_bar_function"]),
+              Attr.on_click(_ =>
+                inject(Update.Set(Breadcrumb_bar(level, true)))
+              ),
+              Attr.on_click(_ =>
+                inject(
+                  UpdateAction.PerformAction(Jump(TileId(List.hd(ids)))),
+                )
+              ),
+            ]),
+          [text(name)],
+        ),
+      ];
+      //if (List.exists(a => a == List.hd(ids), ancestors)) {
+      //} else {
+      //  [];
+      //};
+      let funs_display = Array.make(10, []);
+      funs_display[0] = [];
+      let rec tag_term = (term: TermBase.UExp.t, level: int) => {
+        switch (term.term) {
+        | Let({term: Var(name), _}, {term: Fun(_, fbody), _}, body) =>
+          funs_display[level] =
+            List.cons((name, term.ids), funs_display[level]);
+          tag_term(body, level);
+          tag_term(fbody, level + 1);
+        | UnOp(_, t1)
+        | Fun(_, t1)
+        | TyAlias(_, _, t1)
+        | Test(t1)
+        | Parens(t1)
+        | Ap(_, t1) => tag_term(t1, level)
+        | Let(_, t1, t2)
+        | Seq(t1, t2)
+        | Cons(t1, t2)
+        | ListConcat(t1, t2)
+        | BinOp(_, t1, t2) =>
+          tag_term(t1, level);
+          tag_term(t2, level);
+        | If(t1, t2, t3) =>
+          tag_term(t1, level);
+          tag_term(t2, level);
+          tag_term(t3, level);
+        | ListLit(t)
+        | Tuple(t) => List.iter(t1 => tag_term(t1, level), t)
+        | Match(t1, t) =>
+          tag_term(t1, level);
+          List.iter(t1 => tag_term(t1, level), List.map(snd, t));
+        | _ => ()
+        };
+      };
+      let () = tag_term(term, 1);
+      let rec breadcrumb_funs = (level, res) =>
+        if (level == 0) {
+          res;
+        } else if (funs_display[level] == []) {
+          breadcrumb_funs(level - 1, res);
+        } else {
+          let toggle =
+            List.concat(
+              List.map(
+                t =>
+                  if (List.exists(a => a == List.hd(snd(t)), ancestors)) {
+                    [text(fst(t))];
+                  } else {
+                    [];
+                  },
+                List.rev(funs_display[level]),
+              ),
+            );
+          let siblings =
+            List.concat(
+              List.map(
+                t => view_fun(fst(t), snd(t), level),
+                List.rev(funs_display[level]),
+              ),
+            );
 
+          let siblings_div = {
+            let clss =
+              clss(
+                ["siblings"]
+                @ (
+                  List.nth(settings.breadcrumb_bars, level)
```

Comment:
This will throw an error if someone has 11 nested functions (though anyone writing an 11th nested function probably deserves to have the editor give up on them because their code is too ugly...)

Reply:
After I change into list, it doesn't need a fix integer to limit the nested functions.